FXyz-Core/src/main/java/org/fxyz3d/shapes/primitives/TorusMesh.java

```diff
@@ -1,353 +1,341 @@
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */ 
+
+package org.fxyz3d.shapes.primitives;
+
+import javafx.beans.property.DoubleProperty;
+import javafx.beans.property.IntegerProperty;
+import javafx.beans.property.SimpleDoubleProperty;
+import javafx.beans.property.SimpleIntegerProperty;
+import javafx.scene.DepthTest;
+import javafx.scene.shape.MeshView;
+import javafx.scene.shape.TriangleMesh;
+
+import static java.lang.Math.cos;
+import static java.lang.Math.sin;
+
+/**
+ *
+ * @author jDub1581
+ */
+public class TorusMesh extends MeshView {
+
+    private static final int DEFAULT_DIVISIONS = 64;
+    private static final int DEFAULT_T_DIVISIONS = 64;
+    private static final double DEFAULT_RADIUS = 12.5;
+    private static final double DEFAULT_T_RADIUS = 5.0;
+    private static final double DEFAULT_START_ANGLE = 0.0;
+    private static final double DEFAULT_X_OFFSET = 0.0;
+    private static final double DEFAULT_Y_OFFSET = 0.0;
+    private static final double DEFAULT_Z_OFFSET = 1.0;
+
+    public TorusMesh() {
+        this(DEFAULT_DIVISIONS, DEFAULT_T_DIVISIONS, DEFAULT_RADIUS, DEFAULT_T_RADIUS);
+    }
+
+    public TorusMesh(double radius, double tubeRadius) {
+        this(DEFAULT_DIVISIONS, DEFAULT_T_DIVISIONS, radius, tubeRadius);
+    }
+
+    public TorusMesh(int radiusDivisions, int tubeDivisions, double radius, double tubeRadius) {
+        setRadiusDivisions(radiusDivisions);
+        setTubeDivisions(tubeDivisions);
+        setRadius(radius);
+        setTubeRadius(tubeRadius);
+        
+        setDepthTest(DepthTest.ENABLE);
+        updateMesh();
+    }
+
+    private void updateMesh(){       
+        setMesh(
+                createTorusMesh(
+                        getRadiusDivisions(),
+                        getTubeDivisions(),
+                        (float) getRadius(),
+                        (float) getTubeRadius(),
+                        (float) getTubeStartAngleOffset(),
+                        (float) getxOffset(),
+                        (float) getyOffset(),
+                        (float) getzOffset()
+                )
+        );
+    }
+    
+    private TriangleMesh createTorusMesh(
+            final int radiusDivisions,
+            final int tubeDivisions,
+            final float radius,
+            final float tRadius,
+            final float tubeStartAngle,
+            final float xOffset,
+            final float yOffset,
+            final float zOffset) {
+
+        final int numVerts = tubeDivisions * radiusDivisions;
+        int faceCount = numVerts * 2;
+
+        float[] points = new float[numVerts * 3];
+        float[] texCoords = new float[numVerts * 2];
+        int[] faces = new int[faceCount * 6];
+
+        int pointIndex = 0;
+        int texIndex = 0;
+        int faceIndex = 0;
+
+        float tubeFraction = 1.0f / tubeDivisions;
+        float radiusFraction = 1.0f / radiusDivisions;
+
+        float TWO_PI = 2.0f * 3.141592653589793f;
```

Comment:
It's easier to use `Math.PI` in the previous code to replace the explicit double value (`2 * Math.PI` instead of `2.0f * 3.141592653589793f`) than to create a special `TWO_PI` value, in my opinion.

Reply:
Yes, but I didn't want to change the calculation itself, which used `float`.

Reply:
Then maybe at least use `float TWO_PI = (float) (2 * Math.PI)`?

Reply:
Sure.